packages/core/src/lib/connection_manager.ts

```diff
@@ -0,0 +1,360 @@
+import { Connection } from "@libp2p/interface-connection";
+import { PeerId } from "@libp2p/interface-peer-id";
+import { PeerInfo } from "@libp2p/interface-peer-info";
+import { Peer } from "@libp2p/interface-peer-store";
+import { IRelay, Tags } from "@waku/interfaces";
+import debug from "debug";
+import type { Libp2p } from "libp2p";
+
+import KeepAliveManager, { KeepAliveOptions } from "./keep_alive_manager.js";
+
+const DEFAULT_MAX_BOOTSTRAP_PEERS_ALLOWED = 2;
+const DEFAULT_APPROX_TIME_BEFORE_BOOTSTRAP_FALLBACK_MS = 5000;
+const DEFAULT_MAX_DIAL_ATTEMPTS_FOR_PEER = 3;
+const DEFAULT_LOGGING_INTERVAL = 10_000;
+
+const log = debug("waku:connection-manager");
+
+export interface Libp2pComponents {
+  connectionManager: Libp2p["connectionManager"];
+  peerStore: Libp2p["peerStore"];
+  ping: Libp2p["ping"];
+  dial: Libp2p["dial"];
+}
+
+export interface UpdatedStates {
+  allPeers: Peer[];
+  allDialablePeers: Peer[];
+  bootstrapPeers: Peer[];
+  nonBootstrapPeers: Peer[];
+  dialableBootstrapPeers: Peer[];
+  dialableNonBootstrapPeers: Peer[];
+  allConnections: Connection[];
+  bootstrapConnections: Connection[];
+  nonBootstrapConnections: Connection[];
+}
```

Comment:
This does not seem necessary once you remove the logic to dial more bootstrap peers as a fallback.

Reply:
As discussed here: https://github.com/waku-org/js-waku/pull/1135#discussion_r1089556210 this bloats the code for no functional value.
If the only value is logging, do note that logging at events (peer exchange query triggers, connecting to peer, etc) do provide similar information.